### PR TITLE
Move the changes in PR #20241 to 1.1.x

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject. Please refer #20241.

Fixes ##20241

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by hasithaa** [comment]
@KavinduZoysa We need to rebase this PR with ballerina-1.1.1 branch.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java [hunk 1]
```diff
@@ -363,10 +363,13 @@ private void defineFunction(DataInputStream dataInStream) throws IOException {
...
                         new BAttachedFunction(names.fromString(funcName), invokableSymbol, funcType);
                 BStructureTypeSymbol structureTypeSymbol = (BStructureTypeSymbol) attachedType.tsymbol;
-                structureTypeSymbol.attachedFuncs.add(attachedFunc);
                 if (Names.USER_DEFINED_INIT_SUFFIX.value.equals(funcName)
                         || funcName.equals(Names.INIT_FUNCTION_SUFFIX.value)) {
                     structureTypeSymbol.initializerFunc = attachedFunc;
+                } else if (funcName.equals(Names.GENERATED_INIT_SUFFIX.value)) {
+                    ((BObjectTypeSymbol) structureTypeSymbol).generatedInitializerFunc = attachedFunc;
+                } else {
+                    structureTypeSymbol.attachedFuncs.add(attachedFunc);
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java]: With this change, the initializers are not added to attached functions list. Is that intentional?

Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java]: Yes, initializers are not added to attached function.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 2]
```diff
@@ -348,17 +348,18 @@ private void addAttachedFunctionsToPackageLevel(BLangPackage pkgNode, SymbolEnv
...
                 }
 
-                if (objectTypeNode.initFunction == null) {
-                    objectTypeNode.initFunction = createInitFunctionForStructureType(objectTypeNode, env,
-                                                                                     Names.USER_DEFINED_INIT_SUFFIX);
-                }
-
-                // Add init function to the attached function list
+                objectTypeNode.generatedInitFunction = createGeneratedInitializerFunction(objectTypeNode, env);
                 BObjectTypeSymbol objectSymbol = ((BObjectTypeSymbol) objectTypeNode.type.tsymbol);
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]: Unused var

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 3]
```diff
@@ -367,6 +368,65 @@ private void addAttachedFunctionsToPackageLevel(BLangPackage pkgNode, SymbolEnv
         }
     }
 
+    private BLangFunction createGeneratedInitializerFunction(BLangObjectTypeNode objectTypeNode, SymbolEnv env) {
+
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]:
```suggestion
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 4]
```diff
@@ -367,6 +368,65 @@ private void addAttachedFunctionsToPackageLevel(BLangPackage pkgNode, SymbolEnv
...
+        ((BInvokableType) generatedInitFunc.symbol.type).restType = initializerFunc.type.restType;
+
+        generatedInitializerFunc.type = initializerFunc.type;
+        generatedInitFunc.desugared = false;
+        return generatedInitFunc;
+    }
+
+    private void addRequiredParamsToGeneratedInitFunction(BLangFunction initFunction, BLangFunction generatedInitFunc,
+                                                          BAttachedFunction generatedInitializerFunc) {
+
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]:
```suggestion
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 5]
```diff
@@ -367,6 +368,65 @@ private void addAttachedFunctionsToPackageLevel(BLangPackage pkgNode, SymbolEnv
...
+                                    requiredParameter.symbol.pkgID,
+                                    requiredParameter.type, requiredParameter.symbol.owner));
+            generatedInitFunc.requiredParams.add(var);
+            generatedInitializerFunc.symbol.params.add(var.symbol);
+        }
+    }
+
+    private void addRestParamsToGeneratedInitFunction(BLangFunction initFunction, BLangFunction generatedInitFunc,
+                                                      BAttachedFunction generatedInitializerFunc) {
+
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]:
```suggestion
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java (hunk 2 above)
Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]: Removed

**Comment by KavinduZoysa** [comment]
> Can you remove the unnecessary new lines which seem to have been added due a configuration in the code style settings?

Done

**Pull request merged**